### PR TITLE
Custom note naming

### PR DESCRIPTION
**Pull request by mgramigna**
This PR is for Jira 208, which allows the user to edit the names of unsigned notes, and adds a default name other than just "New Note" which was the previous default.

Added Functionality:
- When the new note button is clicked, the default note name will now be based off of the clinician's name, the date, and current time, e.g. "Dr. X123 Y987-20183007-1106."
- In the note viewer, there is now a button with a pencil icon that will enable editing of the note name
    - When clicked, the note name field will change into a text box where the user can make changes to them.
    - Hitting enter will confirm the edits and the name change will be saved.
    - Signed notes will not have the pencil button, as their names cannot be changed.

**Note**: We discussed enforcing uniqueness of note names, but we are not quite sure if this is actually what is desired based on how the names are created, so this functionality was not implemented on this branch. 

_Pull requests into Flux require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:._

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome
- [x] Manually tested in IE

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- N/A Cheat sheet is updated
- N/A Demo script is updated 
- N/A Documentation on Wiki has been updated 
- N/A Additional technical components and libraries are documented and added to wiki as needed

**NoteParser**

- N/A Note parser has been updated if structured phrases change

**Code Quality**

- [x] 4-space indents - convert any tabs to spaces
- [x] Use public class field syntax for binding functions - ex. handleChange = () => { ... };
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- N/A Added Enzyme-UI tests for slim mode 
- N/A Added Enzyme-UI tests for full mode
- N/A Added backend tests for new functionality added
- N/A Added note parser examples to test new functionality

## Reviewer 1: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

## Reviewer 2: Name

- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: src/notes/FluxNotesEditor.jsx [hunk 1]
```diff
@@ -1146,6 +1148,53 @@ class FluxNotesEditor extends React.Component {
...
+        this.setState({
+            isEditingNoteName: true
+        });
+    }
+
+    /**
+     * Handle the user submitting a new name for a note
+     */
+    submitNoteNameChange = (e) => {
+        if(e.key === "Enter") {
```
Comment (danlee1025) [inline src/notes/FluxNotesEditor.jsx]: Just put a space after the `if` and before the `(`

File: src/notes/FluxNotesEditor.jsx [hunk 2]
```diff
@@ -1146,6 +1148,53 @@ class FluxNotesEditor extends React.Component {
...
+            this.editor.focus();
+        }
+    }
+
+    /**
+     * Render a TextField if the user wishes to edit the note name, otherwise render the note name as plain text
+     */
+    renderNoteNameEditor = (noteTitle, signed) => {
+        let noteTag;
+        if(signed) {
```
Comment (danlee1025) [inline src/notes/FluxNotesEditor.jsx]: Same with this line.

File: src/notes/FluxNotesEditor.jsx [hunk 3]
```diff
@@ -1146,6 +1148,53 @@ class FluxNotesEditor extends React.Component {
...
+            noteTag = <p className="note-description-detail-value" id="note-title">{noteTitle}</p>;
+        } else {
+            noteTag = <p className="note-description-detail-value" id="note-title"><FontAwesome id="edit-note-name-btn" name="pencil" onClick={this.enableNoteNameEditing} />&nbsp;{noteTitle}</p>;
+        }
+        if(this.state.isEditingNoteName) {
```
Comment (danlee1025) [inline src/notes/FluxNotesEditor.jsx]: And this..

**Comment by nicoleng12** [comment]
This looks good! The back end tests pass. I ran through the demo script and that still works as well. I tested this in chrome and IE and everything works. @gregquinn2001  do you want to review this as well since this was a task you were particularly interested in?

**Comment by gregquinn2001** [comment]
If we have 2 reviews, go for a merge! I'll be happy to see this functionality in there! Matt and I discussed it a number of times as he was working on it, but thanks for the opportunity! :)

**Pull request merged**